homeassistant/components/sensor/discogs.py

```diff
@@ -40,32 +43,75 @@
     """Set up the Discogs sensor."""
     import discogs_client
 
-    name = config.get(CONF_NAME)
-    token = config.get(CONF_TOKEN)
+    token = config[CONF_TOKEN]
 
     try:
-        discogs = discogs_client.Client(SERVER_SOFTWARE, user_token=token)
-        identity = discogs.identity()
+        discogs_client = discogs_client.Client(
+            SERVER_SOFTWARE, user_token=token)
     except discogs_client.exceptions.HTTPError:
         _LOGGER.error("API token is not valid")
         return
 
-    async_add_entities([DiscogsSensor(identity, name)], True)
+    async_add_entities([
+        DiscogsCollectionSensor(discogs_client),
+        DiscogsWantlistSensor(discogs_client),
+        DiscogsRandomRecordSensor(discogs_client),
+    ], True)
 
 
-class DiscogsSensor(Entity):
+class DiscogsCollectionSensor(Entity):
     """Get a user's number of records in collection."""
 
-    def __init__(self, identity, name):
-        """Initialize the Discogs sensor."""
-        self._identity = identity
-        self._name = name
+    def __init__(self, client):
+        """Initialize the Discogs collection sensor."""
+        self._client = client
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return NAME_COLLECTION_SENSOR
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        return ICON_RECORD
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit this state is expressed in."""
+        return UNIT_RECORDS
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the sensor."""
+        return {
+            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+            ATTR_IDENTITY: self._client.identity().name,
+        }
+
+    async def async_update(self):
+        """Set state to the amount of records in user's collection."""
+        self._state = self._client.identity().num_collection
+
+
+class DiscogsWantlistSensor(Entity):
```

Comment:
Other option is to add this as attributes to the existing sensor. Value would be available for automation and the hange not invasive for the users.

Reply:
Hmm yeah, but the whole idea is to have different sensors like those for Darksky